### ISSUE TITLE
ENHANCE_YOUR_CALM / too_many_pings

### ISSUE THREAD

**Issue by raphink** [oboukili/terraform-provider-argocd#124]
Upon creating multiple resources (`argocd_project` and `argocd_application`) on the same server, I'm often getting GOAWAY responses:

```
Error: application thanos could not be created
│ 
│   with module.metrics-archives.module.thanos.argocd_application.this,
│   on .terraform/modules/metrics-archives/modules/main.tf line 39, in resource "argocd_application" "this":
│   39: resource "argocd_application" "this" {
│ 
│ rpc error: code = Unknown desc = closing transport due to: connection error: desc = "error reading from server: EOF", received prior goaway: code: ENHANCE_YOUR_CALM, debug data:
│ too_many_pings
```

From what I understand, this is because the Grpc client is sending more pings than the server is configured to receive. Could it be because the resources are applied in parallel?

**Comment by luke-plausin-kidsloop** [comment]
I am also impacted by this. It would be nice if the ArgoCD provider could implement backoff and retry logic when this happens.

**Comment by onematchfox** [comment]
Believe this should have been resolved in this provider (`>v4.0.0`) due to the dependency upgrade in #207, which contained the following [fix](https://github.com/argoproj/argo-cd/pull/9922) to `argoproj/argo-cd`. Feel free to let me know if this is still an issue, and we can look at it again.

**Comment by peturgq** [comment]
Hi I am encountering this issue on 5.6.0, shot in the dark but it seems the provider might be a bit too aggressive with state refreshing.

**Comment by jcogilvie** [comment]
@onematchfox I am also encountering this on 5.6.0 with argo 2.6.7.  

I am creating an argo project, repo, and multi-source application.  The project and app are already created; the biggest problem I have is when adding a new source to the app.  The issue is repeatable.

**Comment by jcogilvie** [comment]
@peturgq I opened #326 to track the new issue.  By chance are you using a multi-source app?

**Comment by HariSekhon** [comment]
@jcogilvie I've got this issue on ArgoCD v2.6.7 and it's not a multi-source app, it's pulling all manifests from the main internal kubernetes repo.
